<a href="https://colab.research.google.com/github/dyou-wbg/meta_processing/blob/main/messy_metadata_TitleBasedRCTclassifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')
import pandas as pd
# as of today
import datetime as dt
dt.date.today()
import os
import pandas as pd
from google.colab import drive
import csv
import os


Mounted at /content/drive


In [ ]:
import csv
import os
data_dir = '/content/drive/MyDrive/finalMetaFile'
files = os.listdir(data_dir)
files


['metadata_jstor.csv',
 'metadata_scopus 2.csv',
 'elsevier_metadata.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv',
 'metadata_econlit.csv',
 '.DS_Store',
 'SSRN_metadata.csv',
 'log.csv',
 'metadata_repec.csv']

In [ ]:



row_counts = {}
for filename in files:
  if filename.endswith('.csv') :
    filepath = os.path.join(data_dir, filename)
    print(filename)
    with open(filepath, 'r') as csvfile:
      reader = csv.reader(csvfile)
      row_count = sum(1 for row in reader)
      row_counts[filename] = row_count

for filename, row_count in row_counts.items():
  print(f'{filename}: {row_count}')

metadata_jstor.csv
metadata_scopus 2.csv
elsevier_metadata.csv
metadata_semantic_scholar.csv
url1_ie_table_complete.csv
metadata_econlit.csv
SSRN_metadata.csv
log.csv
metadata_repec.csv


Error: field larger than field limit (131072)

# 1.Check each file format and data

In [ ]:
df = pd.read_table('/content/drive/MyDrive/finalMetaFile/metadata_repec.tsv', sep='\t', nrows=10)

In [ ]:
df.head()

filename  \
0  http-155-185-68-2-recentpaper-recent-wp127-pdf...   
1              http-155-185-68-2-wpdemb-0168-pdf.pdf   
2  http-193-0-126-150-portalefiles-3842-journal-o...   
3  http-50sea-com-wp-content-uploads-2019-06-ayes...   
4  http-50sea-com-wp-content-uploads-2019-07-e-go...   

                                               title      type  \
0  The Macroeconomics Outcome of Oil Shocks in th...  preprint   
1  Tertiary education decisions of immigrants and...  preprint   
2                      Deep Roots of Fiscal Behavior   article   
3  Estimation of Wheat Area using Sentinel-1 and ...   article   
4  Role of Inter-Organizational Information Integ...   article   

                                             journal     date  \
0                                                NaN  2017-02   
1                                                NaN  2020-03   
2         Journal of Banking and Financial Economics     2014   
3  International Journal of Agriculture and Susta...     2019   
4  International Journal of Innovations in Scienc...     2019   

                                                 url  
0   http://155.185.68.2/Recentpaper/recent-wp127.pdf  
1                http://155.185.68.2/wpdemb/0168.pdf  
2  http://193.0.126.150/portaleFiles/3842-journal...  
3  http://50sea.com/wp-content/uploads/2019/06/Ay...  
4  http://50sea.com/wp-content/uploads/2019/07/E-...

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_jstor.csv', nrows=10)
df2.head()

id  \
0                http://www.jstor.org/stable/1182553   
1                http://www.jstor.org/stable/1030535   
2  http://www.jstor.org/stable/10.5325/transporta...   
3         http://www.jstor.org/stable/10.1086/587909   
4                http://www.jstor.org/stable/1094350   

                                               title  \
0  Economic Education in Our Schools: A Renewed M...   
1                 Can Germany Contain Russia Safely?   
2  Assessing the Impact of Sarbanes-Oxley Act on ...   
3  Childhood Educational Disruption and Later Lif...   
4          The Exclusion of Liability for Negligence   

                                            isPartOf  publicationYear  doi  \
0                  The Journal of Economic Education             1984  NaN   
1  The Annals of the American Academy of Politica...             1952  NaN   
2                             Transportation Journal             2014  NaN   
3                           Journal of Human Capital             2008  NaN   
4                              The Modern Law Review             1974  NaN   

   docType        docSubType provider  collection datePublished  ...  \
0  article  research-article    jstor         NaN    1984-10-01  ...   
1  article  research-article    jstor         NaN    1952-07-01  ...   
2  article  research-article    jstor         NaN    2014-01-01  ...   
3  article  research-article    jstor         NaN    2008-06-01  ...   
4  article  research-article    jstor         NaN    1974-03-01  ...   

                                                 url  \
0                http://www.jstor.org/stable/1182553   
1                http://www.jstor.org/stable/1030535   
2  http://www.jstor.org/stable/10.5325/transporta...   
3         http://www.jstor.org/stable/10.1086/587909   
4                http://www.jstor.org/stable/1094350   

                               creator                        publisher  \
0                     Richard M. Cyert           Taylor & Francis, Ltd.   
1                     Joseph C. Harsch          Sage Publications, Inc.   
2  Mahesh Srinivasan; Akhilesh Chandra      Penn State University Press   
3                          Paul Heaton  The University of Chicago Press   
4                        J. R. M. Lowe                Modern Law Review   

  language pageStart pageEnd  placeOfPublication  wordCount  pageCount  \
0      eng       261     264                 NaN       1824          4   
1      eng        36      40                 NaN       2570          5   
2      eng        44      78                 NaN       9922         35   
3      eng       154     187                 NaN      11759         34   
4      eng       218     221                 NaN       2072          4   

              file  
0  part-1.jsonl.gz  
1  part-1.jsonl.gz  
2  part-1.jsonl.gz  
3  part-1.jsonl.gz  
4  part-1.jsonl.gz  

[5 rows x 22 columns]

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_scopus 2.csv', nrows=10)
df2.head()

Unnamed: 0.1  Unnamed: 0              scopus_ID  \
0             0           0  SCOPUS_ID:85101100172   
1             1           1  SCOPUS_ID:85096201068   
2             2           2  SCOPUS_ID:85056885088   
3             5           5  SCOPUS_ID:84992147324   
4             6           6  SCOPUS_ID:85089158368   

                                         description   coverDate  \
0  © 2021 by ASERS® Publishing. All rights rese...  2021-01-01   
1  © 2020 Elsevier LtdMaritime Knowledge Cluster...  2021-01-01   
2  © 2017, Copyright © Taylor & Francis Group, ...  2017-10-02   
3  © 2016 Elsevier B.V.Changes in residual volat...  2016-12-01   
4  © 2020 Elsevier B.V.This study documents an u...  2020-10-01   

  aggregationType                                                url  \
0         Journal  https://api.elsevier.com/content/abstract/scop...   
1         Journal  https://api.elsevier.com/content/abstract/scop...   
2         Journal  https://api.elsevier.com/content/abstract/scop...   
3         Journal  https://api.elsevier.com/content/abstract/scop...   
4         Journal  https://api.elsevier.com/content/abstract/scop...   

     source-id  citedby-count  \
0  21100317750              0   
1        27851              4   
2  21100886534              3   
3        28976             17   
4        17496              2   

                                               title  openaccess  \
0  The marketing efficiency development to create...         NaN   
1  Maritime knowledge clusters: A conceptual mode...         0.0   
2  From “Communautaire Spirit” to the “Ghos...         1.0   
3    Testing for identification in SVAR-GARCH models         2.0   
4  Move to success? Headquarters relocation, poli...         0.0   

  subtypeDescription                                  publicationName  \
0            Article  Journal of Environmental Management and Tourism   
1            Article                                    Marine Policy   
2            Article       International Journal of Political Economy   
3            Article         Journal of Economic Dynamics and Control   
4            Article                     Journal of Corporate Finance   

                              doi  abstracts language  
0      10.14505/jemt.v12.1(49).23        NaN      eng  
1    10.1016/j.marpol.2020.104299        NaN      eng  
2   10.1080/08911916.2017.1407737        NaN      eng  
3      10.1016/j.jedc.2016.09.007        NaN      eng  
4  10.1016/j.jcorpfin.2020.101698        NaN      eng

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/elsevier_metadata.csv', nrows=10)
df2.head()

Unnamed: 0  id     type                                              title  \
0           0   0  article  Academic versus industry patenting: An in-dept...   
1           1   1  article  Unpacking transnational industry legitimacy dy...   
2           2   2  article  Influencing scientists' collaboration and prod...   
3           3   3  article  New forms of organisation and R&D internationa...   
4           4   4  article  Prospective analysis of technological innovati...   

                                            abstract  \
0                                                NaN   
1  This paper posits that the combination of chan...   
2  This paper analyzes the effect of university r...   
3  This paper explores the strategic internationa...   
4  In this article, we put forward a concept for ...   

                                            keywords  \
0                                                NaN   
1  Catch-up; Complex product systems; Industry le...   
2  University research center Research collaborat...   
3  Food and beverages Multinational enterprises R...   
4  Variation Technological innovation systems Dev...   

                                             authors    date  \
0  Sapsalis, Eleftherios, van Pottelsberghe de la...  2006.0   
1                Kwak, Kiho, Yoon, Hyungseok (David)  2020.0   
2          Ponomariov, Branco L., Boardman, P. Craig  2010.0   
3  Filippaios, Fragkiskos, Papanastassiou, Marina...  2009.0   
4  Markard, Jochen, Stadelmann, Martin, Truffer, ...  2009.0   

                                                 url          journal  \
0  http://www.sciencedirect.com/science/article/p...  research policy   
1  http://www.sciencedirect.com/science/article/p...  research policy   
2  http://www.sciencedirect.com/science/article/p...  research policy   
3  http://www.sciencedirect.com/science/article/p...  research policy   
4  http://www.sciencedirect.com/science/article/p...  research policy   

   downloaded  
0        True  
1        True  
2        True  
3        True  
4        True

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/url1_ie_table_complete.csv', nrows=10)
df2.head()

Unnamed: 0   id                                              title  \
0         0.0  0.0  Impacts of payments for ecosystem services pro...   
1         1.0  1.0  Providing Collateral And Improving Product Mar...   
2         2.0  2.0  Seeing Is Believing? Evidence From An Extensio...   
3         3.0  3.0  Smallholder Access To Weather Securities In In...   
4         4.0  4.0  Sustainability Of Impact Dimensions Of Decline...   

                                              author  \
0  Alix-Garcia, Jennifer; Aronson, Glen; Radeloff...   
1  Casaburi, Lorenzo; Glennerster, Rachel; Suri, ...   
2  Kondylis, Florence; Mueller, Valerie; Zhu, Siyao;   
3  Ceballos, Francisco; R, Isaac Manuel; Robles, ...   
4  Mcniven, Scott; Gilligan, Daniel O; Hotz, Chri...   

                               sector       country  \
0  Agriculture, fishing, and forestry        Mexico   
1  Agriculture, fishing, and forestry  Sierra Leone   
2  Agriculture, fishing, and forestry    Mozambique   
3  Agriculture, fishing, and forestry         India   
4  Agriculture, fishing, and forestry        Uganda   

                                                 URL  pub_time  \
0  https://developmentevidence.3ieimpact.org/sear...      2015   
1  https://developmentevidence.3ieimpact.org/sear...      2014   
2  https://developmentevidence.3ieimpact.org/sear...      2014   
3  https://developmentevidence.3ieimpact.org/sear...      2015   
4  https://developmentevidence.3ieimpact.org/sear...      2016   

                                                 pub  open_access  \
0  ['Publication Details', '3ie Series Report', '...         True   
1  ['Publication Details', '3ie Series Report', '...         True   
2  ['Publication Details', '3ie Grantee Final Rep...         True   
3  ['Publication Details', '3ie Series Report', '...         True   
4  ['Publication Details', '3ie Series Report', '...         True   

                                            category  \
0  ['Intervention: Forest conservation policy', '...   
1  ['Intervention 1: Agricultural market informat...   
2  ['Intervention: Introduction of agricultural e...   
3  ['Intervention: Index insurance + Introduction...   
4  ['Intervention: Crop management training + See...   

                                            abstract  \
0                                        No abstract   
1  The report presents the results from the evalu...   
2  Extension is designed to enable lab-to-farm te...   
3  Weather-based insurance products insure farmer...   
4  We examine the sustainability of the impact of...   

                                              panels  \
0  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
1  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
2  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
3  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
4  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   

                                          source_url  
0  http://3ieimpact.org/evidence-hub/publications...  
1                   https://doi.org/10.23846/ow31180  
2  https://www.3ieimpact.org/sites/default/files/...  
3  https://www.3ieimpact.org/evidence-hub/publica...  
4  http://3ieimpact.org/evidence-hub/publications...

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_econlit.csv', nrows=10)
df2.head()

recordID                                            pdfLink  \
0       141  https://content.ebscohost.com/cds/retrieve?con...   
1       142  https://content.ebscohost.com/cds/retrieve?con...   
2       143  https://content.ebscohost.com/cds/retrieve?con...   
3       144  https://content.ebscohost.com/cds/retrieve?con...   
4       145  https://content.ebscohost.com/cds/retrieve?con...   

                                               plink shortDbName  \
0  https://search.ebscohost.com/login.aspx?direct...         eoh   
1  https://search.ebscohost.com/login.aspx?direct...         eoh   
2  https://search.ebscohost.com/login.aspx?direct...         eoh   
3  https://search.ebscohost.com/login.aspx?direct...         eoh   
4  https://search.ebscohost.com/login.aspx?direct...         eoh   

                                           journal  pub_date  \
0   Capitalism: A Journal of History and Economics  20221201   
1  Central European Journal of Operations Research  20221201   
2  Central European Journal of Operations Research  20221201   
3  Central European Journal of Operations Research  20221201   
4  Central European Journal of Operations Research  20221201   

                                                 doi  \
0                                            1969735   
1  ['2000323', {'@type': 'doi', '#text': '10.1007...   
2  ['2000325', OrderedDict([('@type', 'doi'), ('#...   
3  ['2000326', OrderedDict([('@type', 'doi'), ('#...   
4  ['2000329', {'@type': 'doi', '#text': '10.1007...   

                                               title  \
0  Microcosms and Macrocosms: Capitalism and the ...   
1     Stochastic Radiotherapy Appointment Scheduling   
2  Examining the Balance between Efficiency and R...   
3           The Variable-Width Strip Packing Problem   
4  Constrained Naive Bayes with Application to Un...   

                                              author  \
0                                     Biltoft, C. N.   
1  ['Braune, Roland', 'Gutjahr, Walter J.', 'Vogl...   
2  ['de Arquer, Maria', 'Ponte, Borja', 'Pino, Ra...   
3                 ['Bodis, Attila', 'Csirik, Janos']   
4  ['Blanquero, Rafael', 'Carrizosa, Emilio', 'Ra...   

                                            abstract  Unnamed: 0.1  
0  Using the metaphor of microcosm and macrocosm,...           NaN  
1  When scheduling the starting times for treatme...           NaN  
2  We study a hybrid system where the demand of c...           NaN  
3  Here, we focus on a generalized version of the...           NaN  
4  The Naive Bayes is a tractable and efficient a...           NaN

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/SSRN_metadata.csv', nrows=10)
df2.head()

Unnamed: 0  id                                           doc_name  \
0           0   0  The Conclusion and Termination of the 'First' ...   
1           1   1  Investing in School Readiness: An Analysis of ...   
2           2   2  Science, Specific Knowledge and Total Quality ...   
3           3   3                       To Believe or Not to Believe   
4           5   5  Self Interest, Altruism, Incentives, and Agenc...   

   posted_date                                            authors  \
0  29 Sep 0012                                   Sunita Jogarajan   
1  09 Jan 0017  Nozomi Nakajima, Amer Hasan, Haeil  Jung, Sall...   
2    11-Jul-94                  Karen H. Wruck, Michael C. Jensen   
3    25-Jul-94   Utpal Bhattacharya , Murugappa (Murgie) Krishnan   
4     5-Oct-94                                  Michael C. Jensen   

                                             doc_url  
0  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
1  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
2  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
3  https://papers.ssrn.com/sol3/papers.cfm?abstra...  
4  https://papers.ssrn.com/sol3/papers.cfm?abstra...

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/log.csv', nrows=10)
df2.head()

iteration working_paper_id pdf_name  \
0          1            w0001    w0001   
1          2            w0002    w0002   
2          3            w0003    w0003   
3          4            w0004    w0004   
4          5            w0005    w0005   

                                             pdf_url  \
0  https://www.nber.org/system/files/working_pape...   
1  https://www.nber.org/system/files/working_pape...   
2  https://www.nber.org/system/files/working_pape...   
3  https://www.nber.org/system/files/working_pape...   
4  https://www.nber.org/system/files/working_pape...   

                 retrieved_at                   result  
0  2022-05-27 14:57:01.807231  successfully retrieved!  
1  2022-05-27 14:57:03.603092  successfully retrieved!  
2  2022-05-27 14:57:05.103170  successfully retrieved!  
3  2022-05-27 14:57:06.977062  successfully retrieved!  
4  2022-05-27 14:57:08.292688  successfully retrieved!

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_semantic_scholar.csv', nrows=10)
df2.head()

paper_id  \
0  0000003ee9b8a4ef644842693ba78ce420cdc439   
1  000001d91985c527a5964dda397210f0eae2ecfd   
2  0000047b1466febc0389fc681456ee858e73d179   
3  000005775773f34cd0e89135b3221ffd5a34eea2   
4  000005f7ebe1a860b19ed9b82fd42361be05b7fe   

                                               title  \
0                Agro-Industries for Economic Growth   
1  Sequential Auctions with Supply or Demand Unce...   
2  Comparative performance of IRDP schemes in gen...   
3  Illiquidity in Intermediary Portfolios: Eviden...   
4  Moving Back Home: Insurance Against Labor Mark...   

                                               venue    year  reference_count  \
0  SEDME (Small Enterprises Development, Manageme...  1980.0                0   
1                                                NaN  1997.0                0   
2                                                NaN  1990.0                0   
3                                                NaN  2018.0                0   
4                                                NaN  2008.0               60   

   citation_count publication_date publication_types  \
0               0       1980-06-01    JournalArticle   
1              12              NaN               NaN   
2               0              NaN               NaN   
3               9       2018-01-05               NaN   
4               0              NaN               NaN   

                                     authors   s2_fields_of_study  \
0                                   N. Joshi            Economics   
1                    R. B. Verde;J. Sákovics  Economics;Economics   
2  D. A. Deshmukh;S. Kalyankar;K. D. Rajmane  Economics;Economics   
3                  D. Barth;Phillip J. Monin   Business;Economics   
4                                Greg Kaplan            Economics   

  fields_of_study                                       journal_name  \
0             NaN  SEDME (Small Enterprises Development, Manageme...   
1       Economics                                                NaN   
2       Economics                                                NaN   
3        Business                    Social Science Research Network   
4             NaN                                                NaN   

         mag_id  corpus_id  pub_med_id  pub_med_central_id  arxiv_id  dblp_id  \
0           NaN        NaN         NaN                 NaN       NaN      NaN   
1  2.793010e+09        NaN         NaN                 NaN       NaN      NaN   
2  3.169189e+08        NaN         NaN                 NaN       NaN      NaN   
3  2.763354e+09        NaN         NaN                 NaN       NaN      NaN   
4           NaN        NaN         NaN                 NaN       NaN      NaN   

   acl_id                       doi  
0     NaN  10.1177/0970846419800204  
1     NaN                       NaN  
2     NaN                       NaN  
3     NaN      10.2139/SSRN.3044468  
4     NaN                       NaN

In [ ]:
df2 = pd.read_csv('/content/drive/MyDrive/finalMetaFile/url1_ie_table_complete.csv', nrows=10)
df2.head()


Unnamed: 0   id                                              title  \
0         0.0  0.0  Impacts of payments for ecosystem services pro...   
1         1.0  1.0  Providing Collateral And Improving Product Mar...   
2         2.0  2.0  Seeing Is Believing? Evidence From An Extensio...   
3         3.0  3.0  Smallholder Access To Weather Securities In In...   
4         4.0  4.0  Sustainability Of Impact Dimensions Of Decline...   

                                              author  \
0  Alix-Garcia, Jennifer; Aronson, Glen; Radeloff...   
1  Casaburi, Lorenzo; Glennerster, Rachel; Suri, ...   
2  Kondylis, Florence; Mueller, Valerie; Zhu, Siyao;   
3  Ceballos, Francisco; R, Isaac Manuel; Robles, ...   
4  Mcniven, Scott; Gilligan, Daniel O; Hotz, Chri...   

                               sector       country  \
0  Agriculture, fishing, and forestry        Mexico   
1  Agriculture, fishing, and forestry  Sierra Leone   
2  Agriculture, fishing, and forestry    Mozambique   
3  Agriculture, fishing, and forestry         India   
4  Agriculture, fishing, and forestry        Uganda   

                                                 URL  pub_time  \
0  https://developmentevidence.3ieimpact.org/sear...      2015   
1  https://developmentevidence.3ieimpact.org/sear...      2014   
2  https://developmentevidence.3ieimpact.org/sear...      2014   
3  https://developmentevidence.3ieimpact.org/sear...      2015   
4  https://developmentevidence.3ieimpact.org/sear...      2016   

                                                 pub  open_access  \
0  ['Publication Details', '3ie Series Report', '...         True   
1  ['Publication Details', '3ie Series Report', '...         True   
2  ['Publication Details', '3ie Grantee Final Rep...         True   
3  ['Publication Details', '3ie Series Report', '...         True   
4  ['Publication Details', '3ie Series Report', '...         True   

                                            category  \
0  ['Intervention: Forest conservation policy', '...   
1  ['Intervention 1: Agricultural market informat...   
2  ['Intervention: Introduction of agricultural e...   
3  ['Intervention: Index insurance + Introduction...   
4  ['Intervention: Crop management training + See...   

                                            abstract  \
0                                        No abstract   
1  The report presents the results from the evalu...   
2  Extension is designed to enable lab-to-farm te...   
3  Weather-based insurance products insure farmer...   
4  We examine the sustainability of the impact of...   

                                              panels  \
0  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
1  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
2  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
3  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   
4  ['SECTOR INFORMATION', 'Sector: Agriculture, f...   

                                          source_url  
0  http://3ieimpact.org/evidence-hub/publications...  
1                   https://doi.org/10.23846/ow31180  
2  https://www.3ieimpact.org/sites/default/files/...  
3  https://www.3ieimpact.org/evidence-hub/publica...  
4  http://3ieimpact.org/evidence-hub/publications...

#  2.Get list

In [ ]:
# as of today
import datetime as dt
dt.date.today()

datetime.date(2024, 2, 22)

In [ ]:
import os
import pandas as pd
from google.colab import drive

# Get the path to the directory containing the files
data_dir = '/content/drive/MyDrive/finalMetaFile'

# List all files in the directory
files = os.listdir(data_dir)
files

['metadata_jstor.csv',
 'metadata_scopus 2.csv',
 'elsevier_metadata.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv',
 'metadata_econlit.csv',
 '.DS_Store',
 'SSRN_metadata.csv',
 'log.csv',
 'metadata_repec.csv']

In [ ]:
files_clean=['metadata_jstor.csv',
 'metadata_scopus 2.csv',
 'elsevier_metadata.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv',
 'metadata_econlit.csv',
 'SSRN_metadata.csv',
 'log.csv',
 'metadata_repec.csv']

dfs = []

#for file in files_clean:
#  df = pd.read_csv(os.path.join(data_dir, file), nrows=10)
#  print('------',file,'------')
#  print(df.columns)


for filename in files_clean:
  if filename.endswith('.csv'):
    filepath = os.path.join(data_dir, filename)
    df = pd.read_csv(filepath, nrows=5)
    print('------',filename,'------')
    print(df.columns)


------ metadata_jstor.csv ------
Index(['id', 'title', 'isPartOf', 'publicationYear', 'doi', 'docType',
       'docSubType', 'provider', 'collection', 'datePublished', 'issueNumber',
       'volumeNumber', 'url', 'creator', 'publisher', 'language', 'pageStart',
       'pageEnd', 'placeOfPublication', 'wordCount', 'pageCount', 'file'],
      dtype='object')
------ metadata_scopus 2.csv ------
Index(['Unnamed: 0.1', 'Unnamed: 0', 'scopus_ID', 'description', 'coverDate',
       'aggregationType', 'url', 'source-id', 'citedby-count', 'title',
       'openaccess', 'subtypeDescription', 'publicationName', 'doi',
       'abstracts', 'language'],
      dtype='object')
------ elsevier_metadata.csv ------
Index(['Unnamed: 0', 'id', 'type', 'title', 'abstract', 'keywords', 'authors',
       'date', 'url', 'journal', 'downloaded'],
      dtype='object')
------ metadata_semantic_scholar.csv ------
Index(['paper_id', 'title', 'venue', 'year', 'reference_count',
       'citation_count', 'publication_

In [ ]:
dfs = []

for file in files_clean:
  df = pd.read_csv(os.path.join(data_dir, file), nrows=5)
  output= pd.DataFrame(df.columns)

#  3.Concate all colnames

In [ ]:

def create_dataframe_of_column_names(directory):
  #files = os.listdir(directory)
  files = files_clean
  dataframes = []

  for file in files:
    if file.endswith('.csv') or filename.endswith('.tsv'):
      df = pd.read_csv(os.path.join(directory, file), nrows=5)
      column_names = df.columns

      df_column_names = pd.DataFrame(column_names)
      df_column_names['filename'] = file
      dataframes.append(df_column_names)

  df_all_column_names = pd.concat(dataframes)
  return df_all_column_names

directory = '/content/drive/MyDrive/finalMetaFile'
df_all_column_names = create_dataframe_of_column_names(directory)
print(df_all_column_names)


                  0            filename
0                id  metadata_jstor.csv
1             title  metadata_jstor.csv
2          isPartOf  metadata_jstor.csv
3   publicationYear  metadata_jstor.csv
4               doi  metadata_jstor.csv
..              ...                 ...
3           journal  metadata_repec.csv
4              date  metadata_repec.csv
5               url  metadata_repec.csv
6        Unnamed: 6  metadata_repec.csv
7        Unnamed: 7  metadata_repec.csv

[114 rows x 2 columns]


In [ ]:
df_all_column_names

0            filename
0                id  metadata_jstor.csv
1             title  metadata_jstor.csv
2          isPartOf  metadata_jstor.csv
3   publicationYear  metadata_jstor.csv
4               doi  metadata_jstor.csv
..              ...                 ...
3           journal  metadata_repec.csv
4              date  metadata_repec.csv
5               url  metadata_repec.csv
6        Unnamed: 6  metadata_repec.csv
7        Unnamed: 7  metadata_repec.csv

[114 rows x 2 columns]

In [ ]:
workingdf=df_all_column_names
workingdf

0            filename
0                id  metadata_jstor.csv
1             title  metadata_jstor.csv
2          isPartOf  metadata_jstor.csv
3   publicationYear  metadata_jstor.csv
4               doi  metadata_jstor.csv
..              ...                 ...
3           journal  metadata_repec.csv
4              date  metadata_repec.csv
5               url  metadata_repec.csv
6        Unnamed: 6  metadata_repec.csv
7        Unnamed: 7  metadata_repec.csv

[114 rows x 2 columns]

In [ ]:
workingdf=df_all_column_names

workingdf = workingdf.rename(columns={0: 'colnam'})
workingdf.index.name = 'Num_col'
workingdf=workingdf.reset_index()
workingdf

Num_col           colnam            filename
0          0               id  metadata_jstor.csv
1          1            title  metadata_jstor.csv
2          2         isPartOf  metadata_jstor.csv
3          3  publicationYear  metadata_jstor.csv
4          4              doi  metadata_jstor.csv
..       ...              ...                 ...
109        3          journal  metadata_repec.csv
110        4             date  metadata_repec.csv
111        5              url  metadata_repec.csv
112        6       Unnamed: 6  metadata_repec.csv
113        7       Unnamed: 7  metadata_repec.csv

[114 rows x 3 columns]

In [ ]:
df_final=workingdf

In [ ]:
df_count=pd.DataFrame(df_final.groupby(['filename'])['Num_col'].max()).reset_index()
df_count.sort_values(by=['Num_col'])
#.rename(columns={0: 'colnam'})


filename  Num_col
0              SSRN_metadata.csv        5
2                        log.csv        5
5             metadata_repec.csv        7
1          elsevier_metadata.csv       10
3           metadata_econlit.csv       10
8     url1_ie_table_complete.csv       13
6          metadata_scopus 2.csv       15
7  metadata_semantic_scholar.csv       19
4             metadata_jstor.csv       21

In [ ]:
pd.options.display.max_rows = 200
pd.set_option('display.max_seq_items', None)
df_final.head(100)

Num_col              colnam                       filename
0         0                  id             metadata_jstor.csv
1         1               title             metadata_jstor.csv
2         2            isPartOf             metadata_jstor.csv
3         3     publicationYear             metadata_jstor.csv
4         4                 doi             metadata_jstor.csv
5         5             docType             metadata_jstor.csv
6         6          docSubType             metadata_jstor.csv
7         7            provider             metadata_jstor.csv
8         8          collection             metadata_jstor.csv
9         9       datePublished             metadata_jstor.csv
10       10         issueNumber             metadata_jstor.csv
11       11        volumeNumber             metadata_jstor.csv
12       12                 url             metadata_jstor.csv
13       13             creator             metadata_jstor.csv
14       14           publisher             metadata_jstor.csv
15       15            language             metadata_jstor.csv
16       16           pageStart             metadata_jstor.csv
17       17             pageEnd             metadata_jstor.csv
18       18  placeOfPublication             metadata_jstor.csv
19       19           wordCount             metadata_jstor.csv
20       20           pageCount             metadata_jstor.csv
21       21                file             metadata_jstor.csv
22        0        Unnamed: 0.1          metadata_scopus 2.csv
23        1          Unnamed: 0          metadata_scopus 2.csv
24        2           scopus_ID          metadata_scopus 2.csv
25        3         description          metadata_scopus 2.csv
26        4           coverDate          metadata_scopus 2.csv
27        5     aggregationType          metadata_scopus 2.csv
28        6                 url          metadata_scopus 2.csv
29        7           source-id          metadata_scopus 2.csv
30        8       citedby-count          metadata_scopus 2.csv
31        9               title          metadata_scopus 2.csv
32       10          openaccess          metadata_scopus 2.csv
33       11  subtypeDescription          metadata_scopus 2.csv
34       12     publicationName          metadata_scopus 2.csv
35       13                 doi          metadata_scopus 2.csv
36       14           abstracts          metadata_scopus 2.csv
37       15            language          metadata_scopus 2.csv
38        0          Unnamed: 0          elsevier_metadata.csv
39        1                  id          elsevier_metadata.csv
40        2                type          elsevier_metadata.csv
41        3               title          elsevier_metadata.csv
42        4            abstract          elsevier_metadata.csv
43        5            keywords          elsevier_metadata.csv
44        6             authors          elsevier_metadata.csv
45        7                date          elsevier_metadata.csv
46        8                 url          elsevier_metadata.csv
47        9             journal          elsevier_metadata.csv
48       10          downloaded          elsevier_metadata.csv
49        0            paper_id  metadata_semantic_scholar.csv
50        1               title  metadata_semantic_scholar.csv
51        2               venue  metadata_semantic_scholar.csv
52        3                year  metadata_semantic_scholar.csv
53        4     reference_count  metadata_semantic_scholar.csv
54        5      citation_count  metadata_semantic_scholar.csv
55        6    publication_date  metadata_semantic_scholar.csv
56        7   publication_types  metadata_semantic_scholar.csv
57        8             authors  metadata_semantic_scholar.csv
58        9  s2_fields_of_study  metadata_semantic_scholar.csv
59       10     fields_of_study  metadata_semantic_scholar.csv
60       11        journal_name  metadata_semantic_scholar.csv
61       12              mag_id  metadata_semantic_scholar.csv
62       13           corpus_id  me

# 4.how many meta files have title (or doc_name) in it?

In [ ]:
df_final['flag'] = df_final['colnam'].apply(lambda x: 1 if x in ['title', 'doc_name'] else 0)


In [ ]:
df_final[df_final['flag'] == 1].head(10)

Num_col    colnam                       filename  flag
1    1        title     metadata_jstor.csv             1   
31   9        title     metadata_scopus 2.csv          1   
41   3        title     elsevier_metadata.csv          1   
50   1        title     metadata_semantic_scholar.csv  1   
71   2        title     url1_ie_table_complete.csv     1   
90   7        title     metadata_econlit.csv           1   
96   2        doc_name  SSRN_metadata.csv              1   
107  1        title     metadata_repec.csv             1

In [ ]:
df_final.groupby('filename')['flag'].sum()
df_title_check=pd.DataFrame(df_final.groupby('filename')['flag'].sum()).reset_index()
df_title_check.sort_values(by=['flag'])

filename  flag
2                        log.csv     0
0              SSRN_metadata.csv     1
1          elsevier_metadata.csv     1
3           metadata_econlit.csv     1
4             metadata_jstor.csv     1
5             metadata_repec.csv     1
6          metadata_scopus 2.csv     1
7  metadata_semantic_scholar.csv     1
8     url1_ie_table_complete.csv     1

# 5.Combine all titles

In [ ]:
# get file name without title or doc_name


In [ ]:
#  get a list of filename values if flag value is 1

df_title_check[df_title_check['flag'] == 1]['filename'].tolist()


['SSRN_metadata.csv',
 'elsevier_metadata.csv',
 'metadata_econlit.csv',
 'metadata_jstor.csv',
 'metadata_repec.csv',
 'metadata_scopus 2.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv']

In [ ]:
final_file_list=df_title_check[df_title_check['flag'] == 1]['filename'].tolist()


In [ ]:
files_clean=final_file_list

for file in files_clean:
  df = pd.read_csv(os.path.join(data_dir, file), nrows=5)
  output= pd.DataFrame(df.columns)

In [ ]:
def load_title_or_doc_name_from_csv(directory):

  dataframes = []

  for file in final_file_list:
    if file.endswith('.csv'):
      df = pd.read_csv(os.path.join(directory, file))

      if 'title' in df.columns:
        dataframes.append(df['title'])
      elif 'doc_name' in df.columns:
        dataframes.append(df['doc_name'])

  return dataframes

# Load the dataframes
dataframes = load_title_or_doc_name_from_csv('/content/drive/MyDrive/finalMetaFile')

# Print the dataframes
for df in dataframes:
  print(df.head())


<ipython-input-25-4af298c5c300>:7: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, file))
<ipython-input-25-4af298c5c300>:7: DtypeWarning: Columns (4,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, file))
<ipython-input-25-4af298c5c300>:7: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, file))


0    The Conclusion and Termination of the 'First' ...
1    Investing in School Readiness: An Analysis of ...
2    Science, Specific Knowledge and Total Quality ...
3                         To Believe or Not to Believe
4    Self Interest, Altruism, Incentives, and Agenc...
Name: doc_name, dtype: object
0    Academic versus industry patenting: An in-dept...
1    Unpacking transnational industry legitimacy dy...
2    Influencing scientists' collaboration and prod...
3    New forms of organisation and R&D internationa...
4    Prospective analysis of technological innovati...
Name: title, dtype: object
0    Microcosms and Macrocosms: Capitalism and the ...
1       Stochastic Radiotherapy Appointment Scheduling
2    Examining the Balance between Efficiency and R...
3             The Variable-Width Strip Packing Problem
4    Constrained Naive Bayes with Application to Un...
Name: title, dtype: object
0    Economic Education in Our Schools: A Renewed M...
1                   Can Germany Cont

In [ ]:
df.head(50)

0     Impacts of payments for ecosystem services pro...
1     Providing Collateral And Improving Product Mar...
2     Seeing Is Believing? Evidence From An Extensio...
3     Smallholder Access To Weather Securities In In...
4     Sustainability Of Impact Dimensions Of Decline...
5     The Impact Of Farmer Field Schools On Knowledg...
6     Assessing The Impact Of Farmer Field Schools O...
7     Disentangling The Effects Of Fair Trade On The...
8     A Triple Win? The Impact Of Tanzania's Joint F...
9     Making Networks Work For Policy: Evidence From...
10    Conservation Policies And Labor Markets: Unrav...
11    Impact Assessment Of Coffee Plantation Renewal...
12    Analysis Of Adoption And Impacts Of Improved M...
13    Assessment Of Adoption And Impact Of Rainwater...
14    Enhancing Rice Production In Uganda: Impact Ev...
15    Creating Opportunities For Rural Producers: Im...
16    Does A Cassava Research-For-Development Progra...
17    Impact Of Information And Communication Te

In [ ]:

pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', -1)

df.head(50)


<ipython-input-27-77714ce11cc0>:4: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  pd.set_option('display.max_colwidth', -1)


0     Impacts of payments for ecosystem services programme in Mexico                                                                                                                            
1     Providing Collateral And Improving Product Market Access For Small Holder Farmers: A Randomised Evaluation Of Inventory Credit In Sierra Leone                                            
2     Seeing Is Believing? Evidence From An Extension Network Experiment In Mozambique                                                                                                          
3     Smallholder Access To Weather Securities In India Demand And Impact On Production Decisions                                                                                               
4     Sustainability Of Impact Dimensions Of Decline And Persistence In Adopting A Biofortified Crop In Uganda                                                                                  
5     The Impact Of Farmer Field Sc

In [ ]:
df_title=pd.DataFrame(df)

In [ ]:
len(df_title)

11377

# 6.Filtering on "RCT" papers

In [ ]:

df_title['flag'] = df_title['title'].apply(lambda x: 1 if 'random' in x.lower() else 0)
df_title[df_title["flag"] == 1]


title  \
1      Providing Collateral And Improving Product Market Access For Small Holder Farmers: A Randomised Evaluation Of Inventory Credit In Sierra Leone                                                                                                                                                                     
63     A Cluster-Randomised Controlled Trial To Compare The Effectiveness Of Different Knowledge-Transfer Interventions For Rural Working Equid Users In Ethiopia                                                                                                                                                         
104    Using a Community-Based Early Childhood Development Center as a Platform to Promote Production and Consumption Diversity Increases Children's Dietary Intake and Reduces Stunting in Malawi: A Cluster-Randomized Trial                                                                                            
210    The Impact Of Voucher Coupons On The Uptake Of Fertilizer And Improved Seeds: Evidence From A Randomized Trial In Mozambique                                                                                                                                                                                       
273    The Impact Of The Use Of New Technologies On Farmers' Wheat Yield In Ethiopia: Evidence From A Randomized Controlled Trial                                                                                                                                                                                         
277    A Randomised Evaluation Of The Effects Of An Agricultural Insurance Programme On Rural Households' Behaviour: Evidence From China                                                                                                                                                                                  
318    Cash For Carbon: A Randomized Trial Of Payments For Ecosystem Services To Reduce Deforestation                                                                                                                                                                                                                     
376    Behavioural Insights to Support Increased Consumption of Quality Protein Maize by Young Children: A Cluster Randomised Trial in Ethiopia                                                                                                                                                                           
411    Effect of fish-oil supplementation on breastmilk long-chain polyunsaturated fatty acid concentration: a randomized controlled trial in rural Ethiopia                                                                                                                                                              
424    Evaluating the Efficacy of Knowledge-Transfer Interventions on Animal Health Knowledge of Rural Working Equid Owners in Central Ethiopia: A Cluster-Randomized Controlled Trial                                                                                                                                    
430    Farm Decision Making and Gender: Results from a Randomized Experiment in Ecuador                                                                                                                                                                                                                                   
440    Home gardening improves dietary diversity, a cluster-randomized controlled trial among Tanzanian women                                                                                                                                                                                                             
473    Measuring the Impacts of Saffron Production Promotion Measures on Farmers' Policy Acceptance Probability: A Randomized Conjoint Field Experiment in Herat Province, Afghanistan                             

In [ ]:

df_title[df_title["flag"] == 1].sum()["flag"]


2642

In [ ]:
len(df_title)

11377

In [ ]:
(df_title[df_title["flag"] == 1].sum()["flag"]/len(df_title))*100.00

23.222290586270546

# 7.Load if RCT

In [ ]:
def load_title_or_doc_name_from_csv_with_RCT(directory):
  dataframes = []

  for file in final_file_list:
    if file.endswith('.csv'):
      df = pd.read_csv(os.path.join(directory, file))

      if 'title' in df.columns:
        df_title = df[df['title'].str.contains('random|RCT', case=False, regex=True)]
        dataframes.append(df_title)
      elif 'doc_name' in df.columns:
        df_title = df[df['doc_name'].str.contains('random|RCT', case=False, regex=True)]
        dataframes.append(df_title)

  return dataframes

# Load the dataframes
dataframes = load_title_or_doc_name_from_csv_with_RCT('/content/drive/MyDrive/finalMetaFile')

# Print the dataframes
for df in dataframes:
  print(df.head())


ValueError: Cannot mask with non-boolean array containing NA / NaN values

In [ ]:
final_file_list

['SSRN_metadata.csv',
 'elsevier_metadata.csv',
 'metadata_econlit.csv',
 'metadata_jstor.csv',
 'metadata_repec.csv',
 'metadata_scopus 2.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv']

In [ ]:
final_file_list=['metadata_jstor.csv',
 'metadata_scopus 2.csv',
 'elsevier_metadata.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv',
 'metadata_econlit.csv',
 'SSRN_metadata.csv',
 'log.csv',
 'metadata_repec.csv']

directory='/content/drive/MyDrive/finalMetaFile'

dataframes = []

for file in final_file_list:
    if file.endswith('.csv'):
      df = pd.read_csv(os.path.join(directory, file), nrows=100)

      if 'title' in df.columns:
        df_title = df[df['title'].str.contains('random|RCT', case=False, regex=True)]
        dataframes.append(df_title)
      elif 'doc_name' in df.columns:
        df_title = df[df['doc_name'].str.contains('random|RCT', case=False, regex=True)]
        dataframes.append(df_title)
return dataframes

dataframes

ValueError: Cannot mask with non-boolean array containing NA / NaN values

In [ ]:
final_file_list

['metadata_jstor.csv',
 'metadata_scopus 2.csv',
 'elsevier_metadata.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv',
 'metadata_econlit.csv',
 'SSRN_metadata.csv',
 'log.csv',
 'metadata_repec.csv']

In [ ]:
for file in final_file_list:
    if file.endswith('.csv'):
        df = pd.read_csv(os.path.join(directory, file), nrows=100)

        if  'doc_name' in df.columns:
            print('with_doc_name:',file)
            dataframes.append(df['doc_name'])

with_doc_name: SSRN_metadata.csv


[0     Economic Education in Our Schools: A Renewed M...
 1                    Can Germany Contain Russia Safely?
 2     Assessing the Impact of Sarbanes-Oxley Act on ...
 3     Childhood Educational Disruption and Later Lif...
 4             The Exclusion of Liability for Negligence
                             ...                        
 95                           Toward Better Benchmarking
 96                                           Discussion
 97                                       Correspondence
 98    Hong Kong and Singapore: Twins or Kissing Cous...
 99    The Performance of Multiperiod Managerial Ince...
 Name: title, Length: 100, dtype: object,
 0     The marketing efficiency development to create...
 1     Maritime knowledge clusters: A conceptual mode...
 2     From “Communautaire Spirit” to the “Ghos...
 3       Testing for identification in SVAR-GARCH models
 4     Move to success? Headquarters relocation, poli...
                             ...              

## 7.1 use this

In [ ]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', -1)

directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','SSRN_metadata.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename), nrows=10)
    if "title" in df.columns:
      df['source']=filename
      dataframes.append(df[["title",'source']])
    elif 'doc_name' in df.columns:
      df_doc_name=df[['doc_name']].rename(columns={"doc_name": "title"}, inplace=True)
      dataframes.append(df_doc_name)

df_combined = pd.concat(dataframes, ignore_index=True)



df_combined.head(100)



#df_title['flag'] = df_title['title'].apply(lambda x: 1 if 'random' in x.lower() else 0)
#df_title[df_title["flag"] == 1]


<ipython-input-32-03bce5ea7280>:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_doc_name=df[['doc_name']].rename(columns={"doc_name": "title"}, inplace=True)
<ipython-input-32-03bce5ea7280>:18: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  pd.set_option('display.max_colwidth', -1)


title  \
0   Economic Education in Our Schools: A Renewed Mission                                                                                                               
1   Can Germany Contain Russia Safely?                                                                                                                                 
2   Assessing the Impact of Sarbanes-Oxley Act on the Logistics Industry:                                                                                              
3   Childhood Educational Disruption and Later Life Outcomes: Evidence from Prince Edward County                                                                       
4   The Exclusion of Liability for Negligence                                                                                                                          
5   The Moscow-Peking-Washington Triangle                                                                                                                              
6   Preserving the New Peace: The Case against NATO Expansion                                                                                                          
7   Succession capitaliste et succession familiale: Un modèle économétrique à deux régimes endogènes                                                                   
8   The Role of Export Cropping in Less Developed Countries: Discussion                                                                                                
9   The One Germany                                                                                                                                                    
10  The marketing efficiency development to create value-added for product and service of community-based tourism. Study case for phatthalung province                 
11  Maritime knowledge clusters: A conceptual model and empirical evidence                                                                                             
12  From “Communautaire Spirit” to the “Ghosts of Maastricht”: European Integration and the Rise of Financialization                                           
13  Testing for identification in SVAR-GARCH models                                                                                                                    
14  Move to success? Headquarters relocation, political favoritism, and corporate performance                                                                          
15  Values–limits of economic rationality and imperialism of culture                                                                                                 
16  Job Search Methods and Wages: Are Natives and Immigrants Different?                                                                                                
17  The organization, economics, and policy of scientific research: What we do know and what we don't know - an agenda for research                                    
18  Product turnover: Simultaneous product market entry and exit                                                                                                       
19  Measuring self-control problems: A structural estimation                                                                                                           
20  Academic versus industry patenting: An in-depth analysis of what determines patent value                                                                           
21  Unpacking transnational industry legitimacy dynamics, windows of opportunity, and latecomers’ catch-up in complex product systems                                  
22  Influencing scientists' collaboration and productivity patterns through new institutions: University research centers and scientific and technical human capital   
23  New forms of organisation and R&D internationalisation among the world's 100 largest food and beverages multinationals     

In [ ]:
# prompt: find rows contains string like "random"

import pandas as pd

# Create a sample dataframe
df = pd.DataFrame({'title': ['This is a random paper', 'This is not a random paper', 'Another random paper']})

# Find rows that contain the string "random"
df_random = df[df['title'].str.contains('random', case=False)]

# Print the resulting dataframe
print(df_random)


In [ ]:
for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename), nrows=200)
    if "doc_name" in df.columns:
      df['source']=filename
      df_RCT=df[df['doc_name'].str.contains('random', case=False, regex=True)]
      dataframes.append(df_RCT[["doc_name",'source']])
      print(filename,':',len(df_RCT)/len(df))



SSRN_metadata.csv : 0.005


In [ ]:
dataframes

[Empty DataFrame
 Columns: [title, source]
 Index: [],
 Empty DataFrame
 Columns: [title, source]
 Index: [],
 Empty DataFrame
 Columns: [title, source]
 Index: [],
 Empty DataFrame
 Columns: [title, source]
 Index: [],
                                                                                                                                                                                                                        title  \
 1    Providing Collateral And Improving Product Market Access For Small Holder Farmers: A Randomised Evaluation Of Inventory Credit In Sierra Leone                                                                            
 63   A Cluster-Randomised Controlled Trial To Compare The Effectiveness Of Different Knowledge-Transfer Interventions For Rural Working Equid Users In Ethiopia                                                                
 104  Using a Community-Based Early Childhood Development Center as a Platform to Promote Production and 

## 7.2 now use this

In [ ]:


import pandas as pd
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','SSRN_metadata.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename))
    if "title" in df.columns:
      df['source']=filename
      df_RCT=df[df['title'].str.contains('random', case=False, regex=True, na=False)]
      dataframes.append(df_RCT[["title",'source']])
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))
    elif 'doc_name' in df.columns:
      df_RCT=df[df['doc_name'].str.contains('random', case=False, regex=True, na=False)]
      dataframes.append(df_RCT)
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))

df_combined = pd.concat(dataframes, ignore_index=True)
df_combined.to_csv('/content/drive/MyDrive/RCT_title_all_2.csv')


<ipython-input-3-8714c1c0bf44>:11: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_jstor.csv : 1095398
metadata_jstor.csv : 0.0012589031566608666
metadata_scopus 2.csv : 711784
metadata_scopus 2.csv : 0.0034195767255234735
elsevier_metadata.csv : 242410
elsevier_metadata.csv : 0.0040757394496926695


<ipython-input-3-8714c1c0bf44>:11: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_semantic_scholar.csv : 7912377
metadata_semantic_scholar.csv : 0.0015926945846993894
url1_ie_table_complete.csv : 11377
url1_ie_table_complete.csv : 0.23222290586270547
metadata_econlit.csv : 19121
metadata_econlit.csv : 0.005909732754563046
SSRN_metadata.csv : 99106
SSRN_metadata.csv : 0.0028454382176659335


<ipython-input-3-8714c1c0bf44>:11: DtypeWarning: Columns (4,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_repec.csv : 915771
metadata_repec.csv : 0.0038666871958164214


* metadata_jstor.csv : 0.12% (0.0012589031566608666)
* metadata_scopus 2.csv : 0.3% (0.0034195767255234735)
* elsevier_metadata.csv : 0.41% (0.0040757394496926695)
* metadata_semantic_scholar.csv : 0.15% (0.0015926945846993894)
* url1_ie_table_complete.csv : 23% (0.23222290586270547)
* metadata_econlit.csv : 0.59% (0.005909732754563046)
* SSRN_metadata.csv : 0.28% (0.0028454382176659335)
* metadata_repec.csv : 0.38% (0.0038666871958164214)

In [ ]:


import pandas as pd
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','SSRN_metadata.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename))
    if "title" in df.columns:
      df['source']=filename
      df_RCT=df[df['title'].str.contains('random|RCT', case=False, regex=True, na=False)]
      dataframes.append(df_RCT[["title",'source']])
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))
    elif 'doc_name' in df.columns:
      df_RCT=df[df['doc_name'].str.contains('random|RCT', case=False, regex=True, na=False)]
      dataframes.append(df_RCT)
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))

df_combined = pd.concat(dataframes, ignore_index=True)
df_combined.to_csv('/content/drive/MyDrive/RCT_title_all_2.csv')


<ipython-input-4-8fedeb9da70a>:11: DtypeWarning: Columns (11) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_jstor.csv : 1095398
metadata_jstor.csv : 0.001844078590612727
metadata_scopus 2.csv : 711784
metadata_scopus 2.csv : 0.003908489092196509
elsevier_metadata.csv : 242410
elsevier_metadata.csv : 0.0046408976527370985


<ipython-input-4-8fedeb9da70a>:11: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_semantic_scholar.csv : 7912377
metadata_semantic_scholar.csv : 0.0019137106333532894
url1_ie_table_complete.csv : 11377
url1_ie_table_complete.csv : 0.2345961149687967
metadata_econlit.csv : 19121
metadata_econlit.csv : 0.006014329794466817
SSRN_metadata.csv : 99106
SSRN_metadata.csv : 0.0030876031723609063


<ipython-input-4-8fedeb9da70a>:11: DtypeWarning: Columns (4,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(os.path.join(directory, filename))


metadata_repec.csv : 915771
metadata_repec.csv : 0.004314397376636736



* metadata_jstor.csv : 1095398
* metadata_jstor.csv : 0.001844078590612727
* metadata_scopus 2.csv : 711784
* metadata_scopus 2.csv : 0.003908489092196509
* elsevier_metadata.csv : 242410
* elsevier_metadata.csv : 0.0046408976527370985
* metadata_semantic_scholar.csv : 7912377
* metadata_semantic_scholar.csv : 0.0019137106333532894
* url1_ie_table_complete.csv : 11377
* url1_ie_table_complete.csv : 0.2345961149687967
* metadata_econlit.csv : 19121
* metadata_econlit.csv : 0.006014329794466817
* SSRN_metadata.csv : 99106
* SSRN_metadata.csv : 0.0030876031723609063
* metadata_repec.csv : 915771
* metadata_repec.csv : 0.004314397376636736

In [ ]:


import pandas as pd
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','SSRN_metadata.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename))
    if "title" in df.columns:
      df['source']=filename
      df_RCT=df[df['title'].str.contains('random|RCT|randomized', case=False, regex=True, na=False)]
      dataframes.append(df_RCT[["title",'source']])
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))
    elif 'doc_name' in df.columns:
      df_RCT=df[df['doc_name'].str.contains('random|RCT|randomized', case=False, regex=True, na=False)]
      dataframes.append(df_RCT)
      print(filename,':',len(df))
      print(filename,':',len(df_RCT)/len(df))

df_combined = pd.concat(dataframes, ignore_index=True)
df_combined.to_csv('/content/drive/MyDrive/RCT_title_all_3keywords.csv')


In [ ]:
import pandas as pd
df = pd.read_csv('/content/drive/MyDrive/finalMetaFile/RCT_title_all.csv')
df_out=pd.DataFrame(df)
df_out.shape

(23981, 9)

In [ ]:
# prompt: save dataframe as csv in drive

df_combined.to_csv('/content/drive/MyDrive/RCT_title_all.csv')


In [ ]:
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv','metadata_scopus 2.csv','elsevier_metadata.csv','metadata_semantic_scholar.csv','url1_ie_table_complete.csv','metadata_econlit.csv','log.csv','metadata_repec.csv']

dataframes = []

for filename in final_file_list:
  if os.path.isfile(os.path.join(directory, filename)) and filename.endswith(".csv"):
    df = pd.read_csv(os.path.join(directory, filename), nrows=300)
    if "title" in df.columns:
      df['source']=filename
      df_RCT=df[df['title'].str.contains('random', case=False, regex=True, na=False)]
      dataframes.append(df_RCT[["title",'source']])
      print(filename,':',len(df_RCT)/len(df))


df_combined = pd.concat(dataframes, ignore_index=True)

metadata_jstor.csv : 0.0
metadata_scopus 2.csv : 0.0
elsevier_metadata.csv : 0.0
metadata_semantic_scholar.csv : 0.0
url1_ie_table_complete.csv : 0.02
metadata_econlit.csv : 0.01
metadata_repec.csv : 0.0


In [ ]:
directory = '/content/drive/MyDrive/finalMetaFile'
final_file_list=['metadata_jstor.csv',
 'metadata_scopus 2.csv',
 'elsevier_metadata.csv',
 'metadata_semantic_scholar.csv',
 'url1_ie_table_complete.csv',
 'metadata_econlit.csv',
 'SSRN_metadata.csv',
 'log.csv',
 'metadata_repec.csv']

dataframes = []
for file in final_file_list:
    if file.endswith('.csv'):
        df = pd.read_csv(os.path.join(directory, file), nrows=100)

        if 'title' in df.columns:
            print('with_title:',file)
            dataframes.append(df[['title']])
        elif 'doc_name' in df.columns:
            print('with_doc_name:',file)
            df_doc_name=df[['doc_name']].rename(columns={"doc_name": "title"}, inplace=True)
            dataframes.append(df_doc_name)
dataframes

with_title: metadata_jstor.csv
with_title: metadata_scopus 2.csv
with_title: elsevier_metadata.csv
with_title: metadata_semantic_scholar.csv
with_title: url1_ie_table_complete.csv
with_title: metadata_econlit.csv
with_doc_name: SSRN_metadata.csv


<ipython-input-17-483b403ccd5d>:22: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_doc_name=df[['doc_name']].rename(columns={"doc_name": "title"}, inplace=True)


with_title: metadata_repec.csv


[                                                title
 0   Economic Education in Our Schools: A Renewed M...
 1                  Can Germany Contain Russia Safely?
 2   Assessing the Impact of Sarbanes-Oxley Act on ...
 3   Childhood Educational Disruption and Later Lif...
 4           The Exclusion of Liability for Negligence
 ..                                                ...
 95                         Toward Better Benchmarking
 96                                         Discussion
 97                                     Correspondence
 98  Hong Kong and Singapore: Twins or Kissing Cous...
 99  The Performance of Multiperiod Managerial Ince...
 
 [100 rows x 1 columns],
                                                 title
 0   The marketing efficiency development to create...
 1   Maritime knowledge clusters: A conceptual mode...
 2   From “Communautaire Spirit” to the “Ghos...
 3     Testing for identification in SVAR-GARCH models
 4   Move to success? Headquarters rel

In [ ]:
df = pd.concat(dataframes, axis=1)
df.head()

title  \
0  Economic Education in Our Schools: A Renewed M...   
1                 Can Germany Contain Russia Safely?   
2  Assessing the Impact of Sarbanes-Oxley Act on ...   
3  Childhood Educational Disruption and Later Lif...   
4          The Exclusion of Liability for Negligence   

                                               title  \
0  The marketing efficiency development to create...   
1  Maritime knowledge clusters: A conceptual mode...   
2  From “Communautaire Spirit” to the “Ghos...   
3    Testing for identification in SVAR-GARCH models   
4  Move to success? Headquarters relocation, poli...   

                                               title  \
0  Academic versus industry patenting: An in-dept...   
1  Unpacking transnational industry legitimacy dy...   
2  Influencing scientists' collaboration and prod...   
3  New forms of organisation and R&D internationa...   
4  Prospective analysis of technological innovati...   

                                               title  \
0                Agro-Industries for Economic Growth   
1  Sequential Auctions with Supply or Demand Unce...   
2  Comparative performance of IRDP schemes in gen...   
3  Illiquidity in Intermediary Portfolios: Eviden...   
4  Moving Back Home: Insurance Against Labor Mark...   

                                               title  \
0  Impacts of payments for ecosystem services pro...   
1  Providing Collateral And Improving Product Mar...   
2  Seeing Is Believing? Evidence From An Extensio...   
3  Smallholder Access To Weather Securities In In...   
4  Sustainability Of Impact Dimensions Of Decline...   

                                               title  \
0  Microcosms and Macrocosms: Capitalism and the ...   
1     Stochastic Radiotherapy Appointment Scheduling   
2  Examining the Balance between Efficiency and R...   
3           The Variable-Width Strip Packing Problem   
4  Constrained Naive Bayes with Application to Un...   

                                               title  
0  The Macroeconomics Outcome of Oil Shocks in th...  
1  Tertiary education decisions of immigrants and...  
2                      Deep Roots of Fiscal Behavior  
3  Estimation of Wheat Area using Sentinel-1 and ...  
4  Role of Inter-Organizational Information Integ...

In [ ]:
# prompt: append dataframe in long shape

df_long = df.stack().reset_index(name='value').rename(columns={'level_1': 'filename'})
df_long.head()


(100, 7)

In [ ]:
dataframes[dataframes['title'].str.contains('random|RCT', case=False, regex=True)]

TypeError: list indices must be integers or slices, not str

# 9.Open file by file
Get proportion info
* with_title: metadata_jstor.csv
* with_title: metadata_scopus 2.csv
* with_title: elsevier_metadata.csv
* with_title: metadata_semantic_scholar.csv
* with_title: url1_ie_table_complete.csv
* with_title: metadata_econlit.csv
* with_title: metadata_repec.csv
* with_doct_name: SSRN_metadata.csv


In [ ]:
df = pd.read_csv('/content/drive/MyDrive/finalMetaFile/metadata_repec.csv', nrows=100)
df_out=pd.DataFrame(df)
df_out[['title']].head(30)

title
0   The Macroeconomics Outcome of Oil Shocks in th...
1   Tertiary education decisions of immigrants and...
2                       Deep Roots of Fiscal Behavior
3   Estimation of Wheat Area using Sentinel-1 and ...
4   Role of Inter-Organizational Information Integ...
5   Computation of Temporal Decline to a Vanished ...
6   Identification of Potential Sites for Rice Pla...
7   Study of Heavy Metals (Lead and Magnesium) Str...
8   Lago Grande in Monticchio Area (Southern Italy...
9   Demarcation of Vulnerable Site for YSB Based o...
10  STATE FINANCIAL STABILITY SUPPORTING CONTINUOU...
11  Bonacich Measures as Equilibria in Network Models
12  Layoff Orders and Occupational Mobility via Un...
13  The relationship between distance-to-default a...
14               Remedies for European growth problem
15  Bonacich Network Measures as Minimum Norm Solu...
16  Participation Tax Rates in Finland, Earnedinco...
17                              Mutually Best Matches
18  The Finnish corporate network - empirical find...
19  Self-control problems and personalized regulat...
20   Search in Networks: The Case of Board Interlocks
21  Take-up of joint and individual liability loan...
22  Recursive Clustering Methods for Network Analysis
23  Subgame Perfect Equilibria in Continuous-Time ...
24  Boosting Non-linear Predictabilityof Macroecon...
25           A Cluster and a Search Market CanCoexist
26  Effects of subsidizing the firstemployee - Emp...
27  How Much Should We Trust Regression Discontinu...
28  The Advantage of Incumbents in Coalitional Bar...
29  Mergers and Acquisitions in the Markets for Di...